api4/resolver_channel_member_test.go

```diff
@@ -0,0 +1,355 @@
+				NickName  string `json:"nickname"`
+			} `json:"user"`
+			Roles []struct {
+				ID            string   `json:"id"`
+				Name          string   `json:"Name"`
+				Permissions   []string `json:"permissions"`
+				SchemeManaged bool     `json:"schemeManaged"`
+				BuiltIn       bool     `json:"builtIn"`
+			} `json:"roles"`
+			LastViewedAt     float64         `json:"lastViewedAt"`
+			LastUpdateAt     float64         `json:"lastUpdateAt"`
+			MsgCount         float64         `json:"msgCount"`
+			MentionCount     float64         `json:"mentionCount"`
+			MentionCountRoot float64         `json:"mentionCountRoot"`
+			NotifyProps      model.StringMap `json:"notifyProps"`
+			SchemeGuest      bool            `json:"schemeGuest"`
+			SchemeUser       bool            `json:"schemeUser"`
+			SchemeAdmin      bool            `json:"schemeAdmin"`
+			Cursor           string          `json:"cursor"`
+		} `json:"channelMembers"`
+	}
+
+	t.Run("all", func(t *testing.T) {
+		input := graphQLInput{
+			OperationName: "channelMembers",
+			Query: `
+	query channelMembers {
+	  channelMembers(userId: "me") {
+	  	channel {
+		  	id
+		  	createAt
+		  	updateAt
+		  	type
+		    displayName
+		    name
+		    header
+		    team {
+		    	id
+		    }
+	  	}
+	  	user {
+	  		id
+	  		username
+	  		email
+	  	}
+	  	msgCount
+	  	mentionCount
+	  	mentionCountRoot
+	  	schemeGuest
+	  	schemeUser
+	  	schemeAdmin
+	  	cursor
+	  }
+	}
+	`,
+		}
+
+		resp, err := th.MakeGraphQLRequest(&input)
+		require.NoError(t, err)
+		require.Len(t, resp.Errors, 0)
+		require.NoError(t, json.Unmarshal(resp.Data, &q))
+		assert.Len(t, q.ChannelMembers, 9)
+
+		numPrivate := 0
+		numPublic := 0
+		numOffTopic := 0
+		numTownSquare := 0
+		for _, ch := range q.ChannelMembers {
+			assert.NotEmpty(t, ch.Channel.ID)
+			assert.NotEmpty(t, ch.Channel.Name)
+			assert.NotEmpty(t, ch.Channel.CreateAt)
+			assert.NotEmpty(t, ch.Channel.UpdateAt)
+			if ch.Channel.Type == model.ChannelTypeOpen {
+				numPublic++
+			} else if ch.Channel.Type == model.ChannelTypePrivate {
+				numPrivate++
+			}
+
+			if ch.Channel.DisplayName == "Off-Topic" {
+				numOffTopic++
+			} else if ch.Channel.DisplayName == "Town Square" {
+				numTownSquare++
+			}
+
+			assert.Equal(t, th.BasicUser.Id, ch.User.ID)
+			assert.Equal(t, th.BasicUser.Username, ch.User.Username)
+			assert.Equal(t, th.BasicUser.Email, ch.User.Email)
+
+			assert.False(t, ch.SchemeGuest)
+
+			if ch.Channel.Team.ID == myTeam.Id {
+				assert.True(t, ch.SchemeAdmin)
+			} else {
+				assert.False(t, ch.SchemeAdmin)
+			}
+			assert.True(t, ch.SchemeUser)
+
+			assert.NotEmpty(t, ch.Cursor)
+
+			switch ch.Channel.ID {
+			case th.BasicChannel.Id:
+				assert.Equal(t, float64(2), ch.MsgCount)
+			case ch1.Id:
+				assert.Equal(t, float64(1), ch.MsgCount)
+			}
+		}
+
+		assert.Equal(t, 2, numPrivate)
+		assert.Equal(t, 7, numPublic)
+		assert.Equal(t, 2, numOffTopic)
+		assert.Equal(t, 2, numTownSquare)
```

Comment:
Same here, maybe we can get these numbers with the v4 API and check against those?